internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -420,47 +713,90 @@ func (r FunctionAppFlexConsumptionResource) Create() sdk.ResourceFunc {
+
+			deploymentSaKey := functionAppFlexConsumption.DeploymentStorageAccessKey
+			deploymentSaEndpoint := functionAppFlexConsumption.DeploymentStorageContainerEndpoint
+			storageAuthType := webapps.AuthenticationType(functionAppFlexConsumption.DeploymentStorageAuthType)
+			storageConnStringForFCApp := "DEPLOYMENT_STORAGE_CONNECTION_STRING"
+			storageUai := functionAppFlexConsumption.DeploymentStorageUserAssignedIdentityID
+
+			if !features.FivePointOh() {
+				if deploymentSaEndpoint == "" {
+					deploymentSaEndpoint = functionAppFlexConsumption.StorageContainerEndpoint
+					deploymentSaKey = functionAppFlexConsumption.StorageAccessKey
+					storageAuthType = webapps.AuthenticationType(functionAppFlexConsumption.StorageAuthType)
+					storageUai = functionAppFlexConsumption.StorageUserAssignedIdentityID
+				}
+				if functionAppStorageAccountString == "" {
+					endpoint := strings.TrimPrefix(deploymentSaEndpoint, "https://")
+					if storageNameIndex := strings.Index(endpoint, "."); storageNameIndex != -1 {
+						storageName := endpoint[:storageNameIndex]
+						if functionAppFlexConsumption.StorageAccountUsesMSI {
+							functionAppStorageAccountString = storageName
+						} else {
+							functionAppStorageAccountString = fmt.Sprintf(StorageStringFmt, storageName, deploymentSaKey, *storageDomainSuffix)
+						}
+					} else {
+						return fmt.Errorf("retrieving storage container endpoint error, the expected format is https://storagename.blob.core.windows.net/containername, the received value is %s", functionAppFlexConsumption.StorageContainerEndpoint)
+					}
+				}
+			}
+
 			expandedIdentity, err := identity.ExpandSystemAndUserAssignedMapFromModel(functionAppFlexConsumption.Identity)
 			if err != nil {
 				return fmt.Errorf("expanding `identity`: %+v", err)
 			}
 
-			blobContainerType := webapps.FunctionsDeploymentStorageType(functionAppFlexConsumption.StorageContainerType)
+			blobContainerType := webapps.FunctionsDeploymentStorageType(functionAppFlexConsumption.DeploymentStorageContainerType)
+			if !features.FivePointOh() && blobContainerType == "" {
+				blobContainerType = webapps.FunctionsDeploymentStorageType(functionAppFlexConsumption.StorageContainerType)
+			}
+
 			storageDeployment := &webapps.FunctionsDeployment{
 				Storage: &webapps.FunctionsDeploymentStorage{
 					Type:  &blobContainerType,
-					Value: &functionAppFlexConsumption.StorageContainerEndpoint,
+					Value: &deploymentSaEndpoint,
 				},
 			}
-			storageAuthType := webapps.AuthenticationType(functionAppFlexConsumption.StorageAuthType)
-			storageConnStringForFCApp := "DEPLOYMENT_STORAGE_CONNECTION_STRING"
-			endpoint := strings.TrimPrefix(functionAppFlexConsumption.StorageContainerEndpoint, "https://")
-			var storageString string
+
+			storageAuth := webapps.FunctionsDeploymentStorageAuthentication{
+				Type: &storageAuthType,
+			}
+
+			endpoint := strings.TrimPrefix(deploymentSaEndpoint, "https://")
+			var deploymentSaConncString string
```

Comment:
The name seems a little unconventional, maybe this?
```suggestion
			var deploymentSaConnString string
```